lib/expander.js

```diff
@@ -566,65 +550,55 @@ function expandConditionsOfPOs(predicateobjects) {
       expandFunction(po.conditions);
 
       for (let j = 0; j < po.objects.length; j++) {
-        const o = po.objects[j];
-        // TODO same as in expandSubjects function
-        po.objects[j] = {
-          function: idlabfn + 'trueCondition',
-          parameters: [
-            {
-              parameter: idlabfn + 'strBoolean',
-              value: po.conditions,
-              from: 'function'
-            }, {
-              parameter: idlabfn + 'str',
-              value: o.function ? o : o.value,
-              type: o.type,
-              from: o.function ? 'function' : 'subject'
-            }
-          ],
-          type: o.type
-        };
-
-        if (o.datatype) {
-          po.objects[j].datatype = o.datatype;
-        }
+        po.objects[j] = conditionToFunction(po.conditions, po.objects[j])
       }
 
       delete po.conditions;
     }
   }
 }
 
+function conditionToFunction(conditions, map, defaultType = undefined) {
```

Comment:
Could you add a function description here? Thanks!

Reply:
done!